src/apps/chatter/room/index.js

```diff
@@ -21,15 +21,16 @@ import { ChatBox } from '../components/chatbox'
 import url from '/util/url'
 
 // We'll need to pass our store to `useRequest`
-import store from '/store'
+import store, { getState } from '/store'
 
 import { WEB_URL } from '/consts'
 
 // Here is our page component which will use the `useRequest` hook.
 export function Room ({ id }) {
   const [password, setPassword] = useState('')
   // const [wasBadPassword, setWasBadPassword] = useState(false)
-  const [passwordInput, setPasswordInput] = useState('')
+  const [passwordInput, setPasswordInput] = useState(null)
+  const [passwordWasSaved, setPasswordWasSaved] = useState(false)
```

Comment:
This could be derived from global state?

```
const savedPassword = useMappedState(this.context.store, ({ roomPasswords }) => roomPasswords[id])
```